src/EditorFeatures/Core/Implementation/KeywordHighlighting/AbstractKeywordHighlighter.cs

```diff
@@ -8,19 +8,30 @@
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Highlighting
 {
-    internal abstract class AbstractKeywordHighlighter<TNode> : IHighlighter
-        where TNode : SyntaxNode
+    internal abstract class AbstractKeywordHighlighter<TNode> : AbstractKeywordHighlighter where TNode : SyntaxNode
     {
+        protected sealed override bool HighlightNode(SyntaxNode node) => node is TNode;
+
+        protected sealed override IEnumerable<TextSpan> GetHighlightsForNode(SyntaxNode node, CancellationToken cancellationToken)
+            => GetHighlights((TNode)node, cancellationToken);
+
+        protected abstract IEnumerable<TextSpan> GetHighlights(TNode node, CancellationToken cancellationToken);
+    }
+
+    internal abstract class AbstractKeywordHighlighter : IHighlighter
+    {
+        protected abstract bool HighlightNode(SyntaxNode node);
```

Comment:
a different approach might be having this check done directly with GetHighlightsForNode - ie. calling it TryGetHighlights and returning null if this is not the node we're interested in, but I'm not sure if that's an improvement

Reply:
I agree. That seems a little weird to me too.